src/main/java/net/minecraftforge/client/resource/ReloadRequirements.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2016.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.client.resource;
+
+import java.util.Set;
+
+import net.minecraftforge.fml.relauncher.Side;
+import net.minecraftforge.fml.relauncher.SideOnly;
+
+/**
+ * Holds all required rules for {@link IResourceType}s that should be reloaded.
+ */
+@SideOnly(Side.CLIENT)
+public interface ReloadRequirements
```

Comment:
When I first hear it, it sounds like an enum :smile:

Reply:
What would you suggest for a better name? I was also quite unsure of this. I had originally considered `ReloadParameters`, but I felt like that wasn't descriptive enough.